cognite/client/_api/synthetic_time_series.py

```diff
@@ -55,46 +59,55 @@ def query(
 
         Examples:
 
-            Request a synthetic time series query with direct syntax
+            Request a synthetic time series query with direct syntax:
 
                 >>> from cognite.client import CogniteClient
                 >>> client = CogniteClient()
-                >>> dps = client.time_series.data.synthetic.query(expressions="TS{id:123} + TS{externalId:'abc'}", start="2w-ago", end="now")
+                >>> dps = client.time_series.data.synthetic.query(
+                ...     expressions="ts{id:123} + ts{externalId:'abc'}",
+                ...     start="2w-ago",
+                ...     end="now")
 
             Use variables to re-use an expression:
 
-                >>> vars = {"A": "my_ts_external_id", "B": client.time_series.retrieve(id=1)}
-                >>> dps = client.time_series.data.synthetic.query(expressions="A+B", start="2w-ago", end="now", variables=vars)
+                >>> ts = client.time_series.retrieve(id=123)
+                >>> variables = {"A": ts, "B": "my_ts_external_id"}
+                >>> dps = client.time_series.data.synthetic.query(
+                ...     expressions="A+B", start="2w-ago", end="now", variables=variables)
 
             Use sympy to build complex expressions:
 
                 >>> from sympy import symbols, cos, sin
-                >>> a = symbols('a')
-                >>> dps = client.time_series.data.synthetic.query([sin(a), cos(a)], start="2w-ago", end="now", variables={"a": "my_ts_external_id"}, aggregate='interpolation', granularity='1m', target_unit='temperature:deg_c')
+                >>> x, y = symbols("x y")
+                >>> dps = client.time_series.data.synthetic.query(
+                ...     [sin(x), y*cos(x)],
+                ...     start="2w-ago",
+                ...     end="now",
+                ...     variables={x: "foo", y: "bar"},
+                ...     aggregate="interpolation",
+                ...     granularity="15m",
+                ...     target_unit="temperature:deg_c")
         """
-        if limit is None or limit == -1:
+        if is_unlimited(limit):
             limit = cast(int, float("inf"))
 
-        tasks = []
-        expressions_to_iterate = expressions if isinstance(expressions, SequenceNotStr) else [expressions]
+        if single_expr := not isinstance(expressions, SequenceNotStr):
```

Comment:
More a personal preference, but makes it more readable IMO
```suggestion
        if is_single_expr := not isinstance(expressions, SequenceNotStr):
```